.goreleaser.yaml

```diff
@@ -0,0 +1,67 @@
+builds:
+# clusterctl-aws
+- id: "clusterctl-aws"
+  main: ./cmd/clusterawsadm
+  binary: bin/clusterctl-aws
+  env:
+    - CGO_ENABLED=0
+  ldflags:
+  - -s -w 
+  - -X 'sigs.k8s.io/cluster-api-provider-aws/v2/version.gitMajor={{.Major}}'
+  - -X 'sigs.k8s.io/cluster-api-provider-aws/v2/version.gitMinor={{.Minor}}'
+  - -X 'sigs.k8s.io/cluster-api-provider-aws/v2/version.gitVersion={{.Version}}'
+  - -X 'sigs.k8s.io/cluster-api-provider-aws/v2/version.gitCommit={{.Commit}}'
+  - -X 'sigs.k8s.io/cluster-api-provider-aws/v2/version.gitTreeState={{ if .IsGitDirty }}dirty{{ else }}clean{{ end }}'
+  - -X 'sigs.k8s.io/cluster-api-provider-aws/v2/version.buildDate={{.Date}}'
+  - -X 'sigs.k8s.io/cluster-api-provider-aws/v2/cmd/clusterawsadm/cmd/version.CLIName=clusterctl-aws'
+  goos:
+  - linux
+  - darwin
+  - windows 
+  goarch:
+  - amd64
+  - arm64
+  ignore:
+  - goos: windows
+    goarch: arm64
+
+# clusterawsadm
+- id: "clusterawsadm"
+  main: ./cmd/clusterawsadm
+  binary: bin/clusterawsadm
+  env:
+    - CGO_ENABLED=0
+  ldflags:
+  - -s -w 
+  - -X 'sigs.k8s.io/cluster-api-provider-aws/v2/version.gitMajor={{.Major}}'
+  - -X 'sigs.k8s.io/cluster-api-provider-aws/v2/version.gitMinor={{.Minor}}'
+  - -X 'sigs.k8s.io/cluster-api-provider-aws/v2/version.gitVersion={{.Version}}'
+  - -X 'sigs.k8s.io/cluster-api-provider-aws/v2/version.gitCommit={{.Commit}}'
+  - -X 'sigs.k8s.io/cluster-api-provider-aws/v2/version.gitTreeState={{ if .IsGitDirty }}dirty{{ else }}clean{{ end }}'
+  - -X 'sigs.k8s.io/cluster-api-provider-aws/v2/version.buildDate={{.Date}}'
+  goos:
+  - linux
+  - darwin
+  - windows 
+  goarch:
+  - amd64
+  - arm64
+  ignore:
+  - goos: windows
+    goarch: arm64
```

Comment:
What does ignore do here?
It looks like we are releasing a windows for arm64 at the moment.
[clusterawsadm-windows-arm64.exe](https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases/download/v2.4.1/clusterawsadm-windows-arm64.exe)

Reply:
cc. @kranurag7

Reply:
I think I've never seen windows running on arm. If something like that exists 🙂, I can update it. 
Also, I'm unsure if we also need to support freebsd here.

Reply:
> I think I've never seen windows running on arm.

It's rare, but Microsoft made a couple Surface laptops that support ARM, as well as a RasPi image. See https://learn.microsoft.com/en-us/windows/arm/overview. Since we're already shipping it, probably worth continuing.

I don't think supporting BSD is a priority. We do not currently build for it.

Reply:
+1 to what @nrb said.
I'd personally re-add windows arm64, and leave freeBSD out.
Thanks!

Reply:
Looks like it's still ignoring windows-arm

Reply:
We will be releasing two binaries `clusterawsadm` and `clusterctl-aws`
I removed it only from the first binary and missed it in the second one so I updated the same.  

The other thing I updated is following given goreleaser new release includes this. 
```diff
-  - -X 'sigs.k8s.io/cluster-api-provider-aws/v2/version.gitTreeState={{ if .IsGitDirty }}dirty{{ else }}clean{{ end }}'
+  - -X 'sigs.k8s.io/cluster-api-operator/version.gitTreeState={{.GitTreeState}}'
```